src/metabase/driver/hive_like.clj

```diff
@@ -0,0 +1,149 @@
+  (hx/->timestamp
+   (hsql/call :from_unixtime
+              (hsql/call :unix_timestamp
+                         expr (hx/literal format-str)))))
+
+(defn- trunc-with-format [format-str expr]
+  (str-to-date format-str (date-format format-str expr)))
+
+(defn date
+  "Converts `expr` into a date, truncated to `unit`, using Hive SQL dialect functions"
+  [unit expr]
+  (case unit
+    :default expr
+    :minute (trunc-with-format "yyyy-MM-dd HH:mm" (hx/->timestamp expr))
+    :minute-of-hour (hsql/call :minute (hx/->timestamp expr))
+    :hour (trunc-with-format "yyyy-MM-dd HH" (hx/->timestamp expr))
+    :hour-of-day (hsql/call :hour (hx/->timestamp expr))
+    :day (trunc-with-format "yyyy-MM-dd" (hx/->timestamp expr))
+    :day-of-week (hx/->integer (date-format "u"
+                                            (hx/+ (hx/->timestamp expr)
+                                                  (hsql/raw "interval '1' day"))))
+    :day-of-month (hsql/call :dayofmonth (hx/->timestamp expr))
+    :day-of-year (hx/->integer (date-format "D" (hx/->timestamp expr)))
+    :week (hsql/call :date_sub
+                     (hx/+ (hx/->timestamp expr)
+                           (hsql/raw "interval '1' day"))
+                     (date-format "u"
+                                  (hx/+ (hx/->timestamp expr)
+                                        (hsql/raw "interval '1' day"))))
+    :week-of-year (hsql/call :weekofyear (hx/->timestamp expr))
+    :month (hsql/call :trunc (hx/->timestamp expr) (hx/literal :MM))
+    :month-of-year (hsql/call :month (hx/->timestamp expr))
+    :quarter (hsql/call :add_months
+                        (hsql/call :trunc (hx/->timestamp expr) (hx/literal :year))
+                        (hx/* (hx/- (hsql/call :quarter (hx/->timestamp expr))
+                                    1)
+                              3))
+    :quarter-of-year (hsql/call :quarter (hx/->timestamp expr))
+    :year (hsql/call :year (hx/->timestamp expr))))
+
+(defn date-interval
+  "Returns a SQL expression to calculate a time interval using the Hive SQL dialect"
+  [unit amount]
+  (hsql/raw (format "(NOW() + INTERVAL '%d' %s)" (int amount) (name unit))))
+
+(defn string-length-fn
+  "A SQL function call that returns the string length of `field-key`"
+  [field-key]
+  (hsql/call :length field-key))
+
+;; ignore the schema when producing the identifier
+(defn qualified-name-components
+  "Return the pieces that represent a path to FIELD, of the form `[table-name parent-fields-name* field-name]`.
+   This function should be used by databases where schemas do not make much sense."
+  [{field-name :name, table-id :table_id, parent-id :parent_id}]
+  (conj (vec (if-let [parent (metabase.models.field/Field parent-id)]
+               (qualified-name-components parent)
+               (let [{table-name :name, schema :schema} (db/select-one ['Table :name :schema], :id table-id)]
+                 [table-name])))
+        field-name))
```

Comment:
It looks like this implementation is the same as the one in `metabase.models.field`, but you took out the part that optionally prepends `schema` (it looks like you're still fetching it from the DB, however). Is this because Hive DBs don't have schemas? If that's the case, just make sure the Tables are getting stored in the DB with a `nil` `schema`. If you take a look at the original function it only includes schema as a "name component" if it's truthy. So you should still be able to use that implementation.

Reply:
I think it would be ok to remove this entirely, but doing so breaks three tests in `metabase.query-processor.middleware.parameters.sql-test` for Spark SQL and Drill (everything else passes just fine). The tests are on lines 447, 468, 479. I suspect that those three tests should be updated to use aliasing as well, or something like it, but I'm not sure what they're testing exactly.